src/common/regions.ts

```diff
@@ -116,27 +121,65 @@ const counties: County[] = chain(state_county_map_dataset)
 class RegionDB {
   constructor(private regions: Region[]) {}
 
-  private filterByType(regionType: RegionType) {
-    return this.all().filter(region => region.regionType === regionType);
-  }
-
   findByFipsCode(fipsCode: FipsCode): Region | null {
     const region = this.regions.find(region => region.fipsCode === fipsCode);
     return region || null;
   }
 
+  findStateByUrlParams(stateId: string): State | null {
```

Comment:
Should this be `stateUrlSegment` or something?  `stateId` is a bit general and in our codebase I'd normally assume that's the 2-letter state code.